internal/service/fw/api_pop_regions_data_source.go

```diff
@@ -0,0 +1,108 @@
+package fw
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	bloxoneclient "github.com/infobloxopen/bloxone-go-client/client"
+	"github.com/infobloxopen/bloxone-go-client/fw"
+	"github.com/infobloxopen/terraform-provider-bloxone/internal/flex"
+	"github.com/infobloxopen/terraform-provider-bloxone/internal/utils"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ datasource.DataSource = &PoPRegionsDataSource{}
+
+func NewPoPRegionsDataSource() datasource.DataSource {
+	return &PoPRegionsDataSource{}
+}
+
+// PoPRegionsDataSource defines the data source implementation.
+type PoPRegionsDataSource struct {
+	client *bloxoneclient.APIClient
+}
+
+func (d *PoPRegionsDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_" + "td_pop_regions"
+}
+
+type TDPoPRegionsModelWithFilter struct {
```

Comment:
```suggestion
type AtcfwPoPRegionModelWithFilter struct {
```
Keep the naming consistent. Most of the naming is wrong, but when we change, we can do it all together.

Reply:
Changed